src/services/permissions-checker.js

```
@@ -89,21 +98,27 @@ class PermissionsChecker {
   }
 
   async checkPermissions(
-    renderingId, collectionName, permissionName, permissionInfos, environmentId = undefined,
+    user, collectionName, permissionName, permissionInfos, environmentId = undefined,
   ) {
     const getPermissions = async (forceRetrieve) => this.permissionsGetter.getPermissions(
-      renderingId, collectionName, permissionName, { forceRetrieve, environmentId },
+      user.renderingId, collectionName, permissionName, { forceRetrieve, environmentId },
     );
+
     const isAllowed = async ({ forceRetrieve = false } = {}) => PermissionsChecker._isAllowed(
       await getPermissions(forceRetrieve),
       permissionName,
       permissionInfos,
+      user,
     );
 
     if (await isAllowed() || await isAllowed({ forceRetrieve: true })) {
       return null;
     }
 
+    if (permissionName === 'liveQueries') {
```

Why do you need to throw this specific error here?